### ISSUE TITLE
Test on validation set every n epochs (instead of every epoch)

### ISSUE THREAD

**Issue by munnybearz** [tflearn/tflearn#176]
Hello,

When training a small model over a large number of epochs the validation set tests take the majority of the runtime. It would be very useful to have the validation set run every n epochs instead of every epoch..

eg. if I wanted to train over 1000 epochs I could set a param in DNN.fit() to test validation every 10 or 20 epochs.. it would make it much much faster whilst giving the valuable validation set trends in TensorBoard

**Comment by aymericdamien** [comment]
You can set a number of step to apply validation and cancel the validation per epoch:
```
model.fit(X, Y, snapshot_step=5000, snapshot_epoch=False)
```
This way you can train for a long time before validating

**Comment by munnybearz** [comment]
Perfect thank you!
